index.ts

```diff
@@ -55,11 +58,27 @@ function useCache(): boolean {
  */
 var defaultCompilerOptions: typescript.CompilerOptions = {
     module: typescript.ModuleKind.CommonJS,
-    outDir: path.join(path.sep, "tmp", "typescript-register", process.cwd()),
+    outDir: getCachePath(process.cwd()),
     target: typescript.ScriptTarget.ES5
 };
 
 /**
+ * Returns path to cache for source directory.
+ * @param  {string} directory Directory with source code
+ * @return {string}           Path with all special characters replaced with _ and
+ *                            prepended path to temporary directory 
+ */
+function getCachePath(directory: string): string {
+    var sanitizeOptions = {
+        replacement: "_"
```

Comment:
It replaces any special characters in filename with underscores (default is to remove them completely but I use this just in case). For example, on my machine with `%TEMP%` set to `T:\Temp` path to my project `D:\Projects\memhub\` will be transformed to `t:\Temp\typescript-register\D_\Projects\memhub\`.